### ISSUE TITLE
List is not updating when data source is changed

### ISSUE THREAD

**Issue by andregoncalvesdev** [deanmcpherson/react-native-sortable-listview#17]
List is not updating when data source is changed

On SortableListView, data = { this.state.data }. When i add a list item on another view, the state gets updated and the list dont update.

**Comment by deanmcpherson** [comment]
Hey @andregoncalvesdev can you provide a code example of your issue?

**Comment by superandrew213** [comment]
@andregoncalvesdev I had the same problem. Try my [fork](https://github.com/superandrew213/react-native-sortable-listview) if you still need this. Essentially it just removes:

```
shouldComponentUpdate: function(props) {
     if (props.hovering !== this.props.hovering) return true;
     if (props.active !== this.props.active) return true;
     if (props.rowData.data !== this.props.rowData.data) return true;
     return false;
   },
```

**Comment by deanmcpherson** [comment]
Thanks @superandrew213, the shouldComponentUpdate method is pretty important for bearable performance, especially on longer lists. Perhaps we should add a rowHasChanged property to all users to implement their own row comparisons.

**Comment by superandrew213** [comment]
@deanmcpherson sounds good! I'll submit a PR when I get to it if you haven't done it yet. I just needed it to update for now.

**Comment by CaleyD** [comment]
It looks like this work as expected if you use an immutable collection of immutable data objects

**Comment by deanmcpherson** [comment]
Implemented in 0.1.0
